ADR/adr-0010-namespace-metadata

```diff
@@ -0,0 +1,39 @@
+# ADR-0010 Namespace Metadata
+
+Date: 2022-December-9
+
+## Status
+
+Accepted
+
+## Approvers
+
+* Ann Marie Fred
+* Gorkem Ercan
+
+## Reviewers
+
+## Context
+
+We need metadata on our namespaces to make Stone Soup easier to operate and maintain. Standardizing namespace metadata will make it easier for us to search our logs and metrics across clusters. It will also allow us to enable logging for outgoing network traffic, one of our security requirements.
+
+## Namespace labels
+
+We will apply the following labels to Stone Soup namespaces, to make them easier to find.
+
+- `stonesoup.redhat.com/namespacetype: "controller"` for namespaces containing Stone Soup controllers
+- `stonesoup.redhat.com/namespacetype: "user-workspace-data"` for User workspaces where Applications, Components, and so on are defined
+- `stonesoup.redhat.com/namespacetype: "user-deployments"` for the namespaces where GitOps deploys applications for users
+- `stonesoup.redhat.com/namespacetype: "user-builds"` for the namespaces where the Pipeline Service has PipelineRuns
```

Comment:
As pointed out by @Michkov in this [slack thread](https://coreos.slack.com/archives/C02CTEB3MMF/p1670591996653489) - let's not introduce another group name, mainly when we know that the name is just an internal and temporal one. 
We already use `appstudio.redhat.com` all over the place, so let's stay with it.

Reply:
+1 to keep it as stonesoup.redhat.com

Mention it in the Consequences that we're going to have to rename it again in the future.

Reply:
Also, for Phase 1 - aren't user builds and the workspace data in the same namespace?

Reply:
Shared namespace does not change the labeling, both labels should be applied I suppose.

Reply:
@ralphbean do you really want to change the labels (and group name) every time when the product/project name changes? A proper migration from one name to another can be very painful (even if we do it only for labels). Using a mix of group names within one project can be confusing, so I would rather stay with one group name, even if it's not perfect and can become outdated. 
In an ideal world, we would use a name that is not bound to a project/product name, so any future changes won't affect us.

Reply:
> do you really want to change the labels (and group name)

No, no I do not.

> A proper migration from one name to another can be very painful ... Using a mix of group names within one project can be confusing,

I agree.

>  In an ideal world, we would use a name that is not bound to a project/product name, so any future changes won't affect us.

Oo, why not pursue this now?  Could we use `cloud.redhat.com` or `apps.redhat.com` or `apps.cloud.redhat.com` or `cicd.redhat.com`?

Reply:
Because this went back and forth I'm not inclined to change it right now.